### PR TITLE
bump multichain helm chart to latest

### PR DESCRIPTION
**Pull request by turadg**
## Description
- Upgrade starship to latest helm chart:  https://github.com/cosmology-tech/starship/compare/devnet-0.2.2...devnet-0.2.10
- enabled e2e tests for cosmoshub staking flows

### Security Considerations
none

### Scaling Considerations
none

### Documentation Considerations
none

### Testing Considerations
CI

### Upgrade Considerations
none

### CODE REVIEW

File: multichain-testing/scripts/fetch-starship-chain-info.ts [hunk 1]
```diff
@@ -38,10 +38,6 @@ const backmap = {
 for (const ibcInfo of ibc.data) {
   ibcInfo.chain_1.chain_name = backmap[ibcInfo.chain_1.chain_name];
   ibcInfo.chain_2.chain_name = backmap[ibcInfo.chain_2.chain_name];
-  for (const c of ibcInfo.channels) {
-    // @ts-expect-error XXX bad typedef
-    c.tags.preferred = c.tags.perferred;
-  }
```
Comment (0xpatrickdev) [inline multichain-testing/scripts/fetch-starship-chain-info.ts:44]: I think we can get rid of `L32-L41` altogether?

File: multichain-testing/scripts/install.sh [hunk 2]
```diff
@@ -24,7 +24,7 @@ NAMESPACE=""
 HELM_REPO="starship"
 HELM_CHART="starship/devnet"
 HELM_REPO_URL="https://cosmology-tech.github.io/starship/"
-HELM_CHART_VERSION="0.2.2"
+HELM_CHART_VERSION="0.2.8"
```
Comment (0xpatrickdev) [inline multichain-testing/scripts/install.sh:27]: This release also contains a fix for: https://github.com/cosmology-tech/starship/issues/482

```diff
diff --git a/multichain-testing/config.yaml b/multichain-testing/config.yaml
index a82dbac5e..320274650 100644
--- a/multichain-testing/config.yaml
+++ b/multichain-testing/config.yaml
@@ -71,6 +71,7 @@ chains:
               allow_messages: ['*']
     faucet:
       enabled: true
+      type: starship
     ports:
       rest: 1314
       rpc: 26654
```

```diff
diff --git a/multichain-testing/test/stake-ica.test.ts b/multichain-testing/test/stake-ica.test.ts
index 06e36bbc6..56d064f57 100644
--- a/multichain-testing/test/stake-ica.test.ts
+++ b/multichain-testing/test/stake-ica.test.ts
@@ -103,13 +103,6 @@ const stakeScenario = test.macro(async (t, scenario: StakeIcaScenario) => {
     `address for ${scenario.chain} is valid`,
   );

-  if (scenario.chain === 'cosmoshub') {
-    // see https://github.com/cosmos/cosmjs/pull/1593 for upstream fix
-    t.pass(
-      `SKIP ${scenario.chain}. @cosmjs/faucet does not support ICA address length.`,
-    );
-    return;
-  }
   const { creditFromFaucet, getRestEndpoint } = useChain(scenario.chain);
   const queryClient = makeQueryClient(getRestEndpoint());
```

We should also be able to make the above changes, and test the cosmoshub staking flow. I was running into issues on my local a ~week ago and haven't had time to investigate further.

The starship env was also running slower for me on this revision (was hitting CPU limits I haven't observed before), so I am cautious of immediately approving. Curious to hear how your experience has running this (and the revision we have on master locally).

Comment (0xpatrickdev) [inline multichain-testing/scripts/install.sh:27]: We also need to update the GH workflow: 
```diff
git diff .github/workflows/multichain-e2e.yml
diff --git a/.github/workflows/multichain-e2e.yml b/.github/workflows/multichain-e2e.yml
index f032ba14e..05e82e650 100644
--- a/.github/workflows/multichain-e2e.yml
+++ b/.github/workflows/multichain-e2e.yml
@@ -50,7 +50,7 @@ jobs:
           # uses ghcr.io/agoric/agoric-sdk:dev image (latest master)
           values: ./agoric-sdk/multichain-testing/config.yaml
           port-forward: true
-          version: 0.2.2
+          version: 0.2.6
           timeout: 30m
           namespace: agoric-multichain
```

**Comment by 0xpatrickdev** [comment]
I see you've added `force-integration` so bumping this convo: https://github.com/Agoric/agoric-sdk/pull/9679#discussion_r1672598345

Does it trigger [Multichain E2E](https://github.com/Agoric/agoric-sdk/actions/workflows/multichain-e2e.yml)? It doesn't appear to

File: multichain-testing/scripts/fetch-starship-chain-info.ts (hunk 1 above)
Comment (turadg) [reply multichain-testing/scripts/fetch-starship-chain-info.ts:44]: yes!

File: multichain-testing/scripts/install.sh (hunk 2 above)
Comment (0xpatrickdev) [reply multichain-testing/scripts/install.sh:27]: Seems my computer was having a bad day 🤷‍♂️

**Comment by turadg** [comment]
@mergify rebase

**Review by 0xpatrickdev — APPROVED** [review]
Approving but removed `automerge:rebase` as I added another small commit - PTAL at [7ede1e4](https://github.com/Agoric/agoric-sdk/pull/9697/commits/7ede1e4da66e2e639b54323f6c92a324eacbe01a)

**Comment by 0xpatrickdev** [comment]
What the heck! Getting really weird flakes after [7ede1e4](https://github.com/Agoric/agoric-sdk/pull/9697/commits/7ede1e4da66e2e639b54323f6c92a324eacbe01a) that seem unrelated to the change:

- 2x Override Chain Registry is failing
- 2x  message: 'Cannot parse value of response for path [data,published.wallet.agoric143vmgvmknf64fua9cf0z9wqqtz8hjzzxh8zt9e.current]: {"value":""}', in `auto-stake-it.test.ts`

Not observing either on local, but will remove the commit for now.

<details>
<summary>diff</summary>

```diff
diff --git a/multichain-testing/config.yaml b/multichain-testing/config.yaml
index 320274650..79c24cbf2 100644
--- a/multichain-testing/config.yaml
+++ b/multichain-testing/config.yaml
@@ -18,12 +18,14 @@ chains:
       updateConfig:
         file: scripts/update-config.sh
     faucet:
-      enabled: false
+      enabled: true
+      type: starship
     ports:
       rest: 1317
       rpc: 26657
       exposer: 38087
       grpc: 9090
+      faucet: 8085
     resources:
       cpu: 1
       memory: 4Gi
diff --git a/multichain-testing/test/tools/wallet.test.ts b/multichain-testing/test/tools/wallet.test.ts
index 8667d5cbc..7fcf24e16 100644
--- a/multichain-testing/test/tools/wallet.test.ts
+++ b/multichain-testing/test/tools/wallet.test.ts
@@ -1,5 +1,6 @@
 import anyTest from '@endo/ses-ava/prepare-endo.js';
 import type { TestFn } from 'ava';
+import type { Denom } from '@agoric/orchestration';
 import { makeQueryClient } from '../../tools/query.js';
 import { createWallet } from '../../tools/wallet.js';
 import { sleep } from '../../tools/sleep.js';
@@ -7,41 +8,41 @@ import { commonSetup } from '../support.js';

 const test = anyTest as TestFn<Record<string, never>>;

-const walletScenario = test.macro(async (t, scenario: string) => {
-  const { useChain } = await commonSetup(t);
-
-  const prefix = useChain(scenario).chain.bech32_prefix;
-  const wallet = await createWallet(prefix);
-  const addr = (await wallet.getAccounts())[0].address;
-  t.regex(addr, new RegExp(`^${prefix}1`));
-  t.log('Made temp wallet:', addr);
-
-  const apiUrl = await useChain(scenario).getRestEndpoint();
-  const queryClient = makeQueryClient(apiUrl);
-  t.log('Made query client');
-
-  const { balances } = await queryClient.queryBalances(addr);
-  t.log('Beginning balances:', balances);
-  t.deepEqual(balances, []);
-
-  const { creditFromFaucet } = useChain(scenario);
-  t.log('Requesting faucet funds');
-
-  await creditFromFaucet(addr);
-  // XXX needed to avoid race condition between faucet POST and LCD Query
-  // see https://github.com/cosmology-tech/starship/issues/417
-  await sleep(1000, t.log);
-
-  const { balances: updatedBalances } = await queryClient.queryBalances(addr);
-  const expectedDenom = scenario === 'osmosis' ? 'uosmo' : 'uatom';
-  t.like(updatedBalances, [{ denom: expectedDenom, amount: '10000000000' }]);
-  t.log('Updated balances:', updatedBalances);
-
-  const bondDenom = useChain(scenario).chain.staking?.staking_tokens?.[0].denom;
-  t.truthy(bondDenom, 'bond denom found');
-  const { balance } = await queryClient.queryBalance(addr, bondDenom!);
-  t.deepEqual(balance, { denom: bondDenom, amount: '10000000000' });
+const walletScenario = test.macro({
+  title: (_, chainName: string, denom: Denom) =>
+    `create a wallet and get ${denom} on ${chainName}`,
+  exec: async (t, chainName: string, denom: Denom) => {
+    const { useChain } = await commonSetup(t);
+
+    const prefix = useChain(chainName).chain.bech32_prefix;
+    const wallet = await createWallet(prefix);
+    const addr = (await wallet.getAccounts())[0].address;
+    t.regex(addr, new RegExp(`^${prefix}1`));
+    t.log('Made temp wallet:', addr);
+
+    const apiUrl = await useChain(chainName).getRestEndpoint();
+    const queryClient = makeQueryClient(apiUrl);
+    t.log('Made query client');
+
+    const { balances } = await queryClient.queryBalances(addr);
+    t.log('Beginning balances:', balances);
+    t.deepEqual(balances, []);
+
+    const { creditFromFaucet } = useChain(chainName);
+    t.log('Requesting faucet funds');
+    await creditFromFaucet(addr, denom);
+    // XXX needed to avoid race condition between faucet POST and LCD Query
+    // see https://github.com/cosmology-tech/starship/issues/417
+    await sleep(1000, t.log);
+
+    const { balance } = await queryClient.queryBalance(addr, denom);
+    t.log('Ending balance:', balance);
+    t.deepEqual(balance, { denom, amount: '10000000000' });
+  },
 });

-test('create a wallet and get tokens (osmosis)', walletScenario, 'osmosis');
-test('create a wallet and get tokens (cosmoshub)', walletScenario, 'cosmoshub');
+test(walletScenario, 'osmosis', 'uosmo');
+test(walletScenario, 'cosmoshub', 'uatom');
+test(walletScenario, 'agoric', 'ubld');
+test(walletScenario, 'agoric', 'uist');
+test(walletScenario, 'osmosis', 'uion');
```
</details>

**Pull request merged**